src/MICore/Transports/LocalUnixTerminalTransport.cs

```diff
@@ -121,6 +124,8 @@ public override void Close()
             // If we are shutting down before the _exitReader has read a line it's possible that
             // there is a thread blocked doing a read() syscall. 
             ForceDisposeStreamReader(_exitReader);
+            _streamReadPidCancellationTokenSource.Cancel();
+            _streamReadPidCancellationTokenSource.Dispose();
```

Comment:
I just tried this, and no, we don't hit `Close()` or anything for that matter.

Any suggestions?